src/main/java/io/swagger/oas/inflector/validators/StringTypeValidator.java

```diff
@@ -57,42 +46,62 @@ public void validate(Object argument, RequestBody body, Iterator<Validator> chai
-                            if (argument instanceof DateTime) {
-                                // TODO
-                            } else if (argument instanceof LocalDate) {
-                                // TODO
-                            } else {
-                                throw new ValidationException()
-                                        .message(new ValidationMessage()
-                                                .code(ValidationError.INVALID_FORMAT)
-                                                .message( " parameter `" +  " value `" + argument + "` is not a valid " + mediaType.getSchema().getFormat()));
-                            }
-                        }
+                    Set<String> allowable = validateAllowedValues(argument, mediaType.getSchema());
+                    if(allowable != null){
+                        throw new ValidationException()
+                            .message(new ValidationMessage()
+                                    .code(ValidationError.UNACCEPTABLE_VALUE)
+                                    .message(" parameter  value `" + argument + "` is not in the allowable values `" + allowable + "`"));
                     }
+                    if (validateFormat(argument,mediaType.getSchema())){
+                        throw new ValidationException()
+                            .message(new ValidationMessage()
+                                    .code(ValidationError.INVALID_FORMAT)
+                                    .message( " parameter value `" + argument + "` is not a valid " + mediaType.getSchema().getFormat()));
+                    }
+
+
                 }
             }
         }
+
         if(chain.hasNext()) {
             chain.next().validate(argument, body, chain);
             return;
         }
 
         return;
     }
+
+    public Set<String> validateAllowedValues(Object argument, Schema schema){
+        if (argument != null && schema != null) {
+            if (schema.getEnum() != null && schema.getEnum().size() > 0) {
+                List<?> values = schema.getEnum();
+                Set<String> allowable = new LinkedHashSet<>();
+                for (Object obj : values) {
+                    allowable.add(obj.toString());
+                }
+                if (!allowable.contains(argument.toString())) {
+                    return allowable;
+                }
+            }
+        }
+        return  null;
+    }
+
+    public boolean validateFormat(Object argument, Schema schema) throws ValidationException{
+        if (argument != null && schema != null) {
+            if ("string".equals(schema.getType()) && ("date".equals(schema.getFormat()) || "date-time".equals(schema.getFormat()))) {
+                if (argument instanceof DateTime) {
+                    // TODO
```

Comment:
is this something planned to be done before PR?

Reply:
removed